Student Alcohol consumption in Portugal:
Context : to study alcohol consumption in young people and the effects it has on their academic performance

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

1 - Introducing the dataset

In [2]:
student = pd.read_csv("../data/raw/student/student-por.csv")

In [3]:
student.head()

school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher  ...   
1     GP   F   17       U     GT3       T     1     1  at_home     other  ...   
2     GP   F   15       U     LE3       T     1     1  at_home     other  ...   
3     GP   F   15       U     GT3       T     4     2   health  services  ...   
4     GP   F   16       U     GT3       T     3     3    other     other  ...   

  famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0      4        3      4     1     1      3        4   0  11  11  
1      5        3      3     1     1      3        2   9  11  11  
2      4        3      2     2     3      3        6  12  13  12  
3      3        2      2     1     1      5        0  14  14  14  
4      4        3      2     1     2      5        0  11  13  13  

[5 rows x 33 columns]

In [4]:
student.shape

(649, 33)

# Alcohol consumption level variable

In [5]:
# weighted average of how much the students drink during weekdays and weekends:
student.rename(columns={'sex':'gender'}, inplace=True)

In [6]:
student['alcohol_index'] = (5*student['Dalc'] + 2*student['Walc'])/7

In [7]:
student.head()

school gender  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  \
0     GP      F   18       U     GT3       A     4     4  at_home   teacher   
1     GP      F   17       U     GT3       T     1     1  at_home     other   
2     GP      F   15       U     LE3       T     1     1  at_home     other   
3     GP      F   15       U     GT3       T     4     2   health  services   
4     GP      F   16       U     GT3       T     3     3    other     other   

   ... freetime goout  Dalc  Walc  health absences  G1  G2  G3 alcohol_index  
0  ...        3     4     1     1       3        4   0  11  11      1.000000  
1  ...        3     3     1     1       3        2   9  11  11      1.000000  
2  ...        3     2     2     3       3        6  12  13  12      2.285714  
3  ...        2     2     1     1       5        0  14  14  14      1.000000  
4  ...        3     2     1     2       5        0  11  13  13      1.285714  

[5 rows x 34 columns]

In [8]:
# Alcohol consumption level based on acl <= 2
student['acl']= student['alcohol_index'] <= 2

In [9]:
# Map the scl to Low and High
student['acl'] = student['acl'].map({True:'Low', False: 'High'})

In [10]:
student.head()

school gender  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  \
0     GP      F   18       U     GT3       A     4     4  at_home   teacher   
1     GP      F   17       U     GT3       T     1     1  at_home     other   
2     GP      F   15       U     LE3       T     1     1  at_home     other   
3     GP      F   15       U     GT3       T     4     2   health  services   
4     GP      F   16       U     GT3       T     3     3    other     other   

   ... goout Dalc  Walc  health  absences  G1  G2  G3 alcohol_index   acl  
0  ...     4    1     1       3         4   0  11  11      1.000000   Low  
1  ...     3    1     1       3         2   9  11  11      1.000000   Low  
2  ...     2    2     3       3         6  12  13  12      2.285714  High  
3  ...     2    1     1       5         0  14  14  14      1.000000   Low  
4  ...     2    1     2       5         0  11  13  13      1.285714   Low  

[5 rows x 35 columns]

# Divide students into groups based on the variable alcohol_index
 - students for which alcohol consumption is less or equal to 2 will be in the group of low alcolhol consumption 
 - students for which alcohol consumption is superior to 2 will be in the group of high alcolhol consumption 

# Confidence intervals

In [11]:
sample_size = student.shape[0]
print(sample_size)

649


We have 649 observations, and because the sample size is greater than 30, we can use the central limit theorem to calculate confidence intervals. The statistical theory tells us that we can calculate confidence intervals for the mean of a variable using the normal distribution. To get a confidence interval, we just need the following three numbers:
- The sample mean of the variable that we are interested in
- The standard error for the formula: the standard, and the sample standard deviation divided by the square root of the sample size
- The confidence level

3 - Performing probability calculations